apps/vmq_server/src/vmq_cluster.erl

```diff
@@ -169,12 +176,32 @@ check_ready([Node|Rest], Acc) ->
     ok = vmq_cluster_node_sup:ensure_cluster_node(Node),
     %% We should only say we're ready if we've established a
     %% connection to the remote node.
-    IsReady1 = IsReady andalso vmq_cluster_node_sup:is_reachable(Node),
+    S = vmq_cluster_node_sup:node_status(Node),
+    IsReady1 = IsReady andalso lists:member(S, [up, init]),
     check_ready(Rest, [{Node, IsReady1}|Acc]);
 check_ready([], Acc) ->
-    ClusterReady =
-    case lists:keyfind(false, 2, Acc) of
-        false -> true;
-        _ -> false
+    OldObj =
+    case ets:lookup(?VMQ_CLUSTER_STATUS, ready) of
+        [] -> {true, 0, 0};
+        [{ready, Obj}] -> Obj
+    end,
+    NewObj =
+    case {all_nodes_alive(Acc), OldObj} of
+        {true, {true, 0, 0}} ->
+            % initial case, possibly single-node cluster
+            {true, 0, 0};
+        {true, {false, NetsplitDetectedCnt, NetsplitResolvedCnt}} ->
+            % Cluster was inconsistent, netsplit resolved
+            {true, NetsplitDetectedCnt, NetsplitResolvedCnt + 1};
+        {false, {true, NetsplitDetectedCnt, NetsplitResolvedCnt}} ->
+            % Cluster was consistent, but isn't anymore
+            {false, NetsplitDetectedCnt + 1, NetsplitResolvedCnt};
+        {false, {false, NetsplitDetectedCnt, NetsplitResolvedCnt}} ->
+            % Cluster was inconsistent, is still inconsistent
+            {false, NetsplitDetectedCnt, NetsplitResolvedCnt}
     end,
-    ets:insert(?VMQ_CLUSTER_STATUS, [{ready, ClusterReady}|Acc]).
+    ets:insert(?VMQ_CLUSTER_STATUS, [{ready, NewObj}|Acc]).
+
+-spec all_nodes_alive([{NodeName::atom(), IsReady::boolean()}]) -> boolean().
+all_nodes_alive(Nodes) ->
+    not lists:keymember(false, 2, Nodes).
```

Comment:
This seems kind of brittle to me, wouldn't it be better to replace with sth like `lists:all(fun({_, true, _}) -> true; (_) -> false end, Nodes)`.